stat.go

```diff
@@ -648,6 +648,47 @@ func KullbackLeibler(p, q []float64) float64 {
 	return kl
 }
 
+// OrdinaryLinearRegression computes the simple linear regression of x and y.
+// It returns the values for \hat{\alpha}, \hat{\beta} and the coefficient
+// of determination, r. If origin is true, the regression is forced to pass
+// through the origin.
+func OrdinaryLinearRegression(x, y, weights []float64, origin bool) (alpha, beta, r float64) {
```

Comment:
document that `OrdinaryLinearRegression` will panic if `x` and `y` have different lengths.
that test is done explicitly in `Covariance`, when `origin==false` and implicitly when `origin==true` at line 664.
perhaps the test should be done explicitly in `OrdinaryLinearRegression` as well.

document also that `weights` can be `nil` and if not `nil` then it should have the same length than `x` and `y` ?